docs/stdlib/set.rst

```diff
@@ -288,7 +292,7 @@ Sets
 
     :index: is type intersection
 
-    Filter the set based on type.
+    Filters a set based on its type. Will return back :eql:type:`anytype`.
```

Comment:
```suggestion
    Filters a set based on its type. Will return back the specified type.
```
It is important that this operator changes the type of the returned set to whatever the filter says. This allows you to access the links and properties specific to that new type.